File: modules/expo-room-scanner/ios/ExpoRoomScannerModule.swift
import ExpoModulesCore
import RoomPlan

public class ExpoRoomScannerModule: Module {
  public func definition() -> ModuleDefinition {
    Name("ExpoRoomScanner")

    // Capability gate: RoomPlan requires iOS 16+ and a LiDAR-equipped device.
    // JS uses this to decide whether to show the "Scan Room" entry point at all.
    Function("isSupported") { () -> Bool in
      if #available(iOS 16.0, *) {
        return RoomCaptureSession.isSupported
      }
      return false
    }

    View(ExpoRoomScannerView.self) {
      Events("onCaptureFinish", "onCaptureError")

      Prop("isScanning") { (view: ExpoRoomScannerView, isScanning: Bool) in
        view.setScanning(isScanning)
      }
    }
  }
}

File: modules/expo-room-scanner/android/src/main/java/expo/modules/roomscanner/ExpoRoomScannerModule.kt
package expo.modules.roomscanner

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

// RoomPlan is an Apple-only framework (ARKit LiDAR). There is no Android equivalent,
// so this module always reports unsupported — the JS side hides the feature entirely.
class ExpoRoomScannerModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("ExpoRoomScanner")

    Function("isSupported") { false }

    View(ExpoRoomScannerView::class) {
      Events("onCaptureFinish", "onCaptureError")
      Prop("isScanning") { _: ExpoRoomScannerView, _: Boolean -> }
    }
  }
}
